### ISSUE TITLE
WCAG 2.2: 2.4.13 Focus appearance

### ISSUE THREAD

**Issue by Febakke** [statisticsnorway/ssb-component-library#975]
**Note**
When referincing WCAG SC 2.4.13 Focus Appearance (AAA) have in mind that this is a part of the latest version of WCAG guidelines and not a part of the Norwegian required guidelines yet. But as we have been working with our focus indicator in Digdirs designsystem with the new requirments in mind lately I thought I could share some input here.

UUTilsynets interpretation of the current required guideline focus visible is: "Hvis fokusmarkeringen er vanskelig å oppfatte for brukeren, er kravet etter tilsynets vurdering ikke oppfylt." [Source](https://www.uutilsynet.no/wcag-standarden/247-synlig-fokus-niva-aa/109)

**Describe the bug**
Focus on checked checkboxes are hard to see. The contrast between the focused and unfocused pixels is actually 3.32:1. This is good enough when looking at the WCAG 2.2 new guidelines on [Focus appearance](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html). But I have to say this is quite surprising as a red/green colorblind person I am having trouble seeing a visible change when it gets focus. 

<img width="160" alt="Skjermbilde 2024-02-23 kl  09 47 24" src="https://github.com/statisticsnorway/ssb-component-library/assets/33222679/6923cf22-adc4-41a0-b169-a5dc643f14d2">

I would recomend using the same focus style as you are using on buttons. By having a border on the outside with an ofset. 

<img width="217" alt="Skjermbilde 2024-02-23 kl  11 31 05" src="https://github.com/statisticsnorway/ssb-component-library/assets/33222679/5f0b878f-d798-43ec-9dfd-883292d15b3c">

**Input fields**
You could also enhance the focus on inputfields. Your focus as of now is to change color from #274247 to #00824D on the border. Those colors have a contrast of 2.20:1. This is lower then the 3:1 threshold described in focus appearance. You also add an extra border outside with 1px width. Since the first border dont have enough contrast to the unfocused pixels you only have 1px border change from unfocus to focused. Here I would also recommend using the same focus as button. Having the focus indicator outside of the component with an offset. (and at least 2px border)

**links**
I really like the link focus style you have created. A small improvement here is to add a `prefered-reduced-motion` as it do not seem to remove / or slow down animation based on my setting being reduce motion. [Docs](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)

**Universal focus indicator**
We are working on an [universal focus indicator](https://www.sarasoueidan.com/blog/focus-indicators/#a-%E2%80%98universal%E2%80%99-focus-indicator) for our designsystem. That should be able to pass all relevant WCAG guidelines both light and dark background without having to change our indicator. Having a bold and consistent indicator that is easy to follow for keyboard users that is easy to implement on most components. (Some exceptions like links for example where we probably will handle it similar to your solution). If you have any input or feedback feel free to leave a comment on our issue here: [#1409](https://github.com/digdir/designsystemet/issues/1409)
